packages/hash/frontend/src/components/hooks/useCreatePage.ts

```diff
@@ -27,8 +30,56 @@ export const useCreatePage = () => {
     ],
   });
 
+  const [setParentPageFn] = useMutation<
+    SetParentPageMutation,
+    SetParentPageMutationVariables
+  >(setParentPage, {
+    refetchQueries: ({ data }) => [
+      {
+        query: getAccountPages,
+        variables: { accountId: data.setParentPage.accountId },
+      },
+    ],
+  });
+
+  const createUntitledPage = async () => {
+    const response = await createPageFn({
+      variables: { accountId, properties: { title: "Untitled" } },
+    });
+
+    const { accountId: pageAccountId, entityId: pageEntityId } =
+      response.data?.createPage ?? {};
+
+    if (pageAccountId && pageEntityId) {
+      return router.push(`/${pageAccountId}/${pageEntityId}`);
+    }
+  };
+
+  const createSubPage = async (parentPageEntityId: string) => {
+    const response = await createPageFn({
+      variables: { accountId, properties: { title: "Untitled" } },
+    });
+
+    const { accountId: pageAccountId, entityId: pageEntityId } =
+      response.data?.createPage ?? {};
+
+    if (pageAccountId && pageEntityId) {
+      await setParentPageFn({
+        variables: {
+          accountId: pageAccountId,
+          pageEntityId,
+          parentPageEntityId,
+        },
+      });
+
+      return router.push(`/${pageAccountId}/${pageEntityId}`);
+    }
+  };
+
   return {
     create: createPageFn,
```

Comment:
`createPageFn` is only used by `createUntitledPage` and `createSubpage`. I return it because I assumed it might be needed at a later point and the dev might not need the redirect functionality. But I guess the functionality might also be needed. I'd remove it from the list of methods returned for now. When it's needed, it can be added back.